File: ANDROID-APPS/LamseyBets/app/src/main/java/dvlp/lamseybets/CheckInternetConnection.java
package dvlp.lamseybets;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

class CheckInternetConnection {
    public CheckInternetConnection(Context context) {

    }

    public static boolean checkConnection(Context context) {
        final ConnectivityManager connMgr = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);

        NetworkInfo activeNetworkInfo = connMgr.getActiveNetworkInfo();

        if (activeNetworkInfo != null) { // connected to the internet


            // connected to the mobile provider's data plan
            if (activeNetworkInfo.getType() == ConnectivityManager.TYPE_WIFI) {
                // connected to wifi
                return true;
            } else return activeNetworkInfo.getType() == ConnectivityManager.TYPE_MOBILE;
        }
        return false;
    }
}

File: ANDROID-APPS/LamseyBets/app/src/main/java/dvlp/lamseybets/CheckInternetConnection.kt
package dvlp.lamseybets

import android.content.Context
import android.net.ConnectivityManager

internal class CheckInternetConnection(context: Context?) {
    companion object {

        fun checkConnection(context: Context): Boolean {
            val connMgr = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager

            val activeNetworkInfo = connMgr.activeNetworkInfo

            return if (activeNetworkInfo != null) { // connected to the internet


                // connected to the mobile provider's data plan
                if (activeNetworkInfo.type == ConnectivityManager.TYPE_WIFI) {
                    // connected to wifi
                    true
                } else
                    activeNetworkInfo.type == ConnectivityManager.TYPE_MOBILE
            } else false
        }
    }
}
